backend/apps/volontulo/permissions.py

```diff
@@ -13,8 +13,17 @@ class OfferPermission(permissions.BasePermission):
 
     def has_permission(self, request, view):
         """We are accepting only safe methods for now."""
+        if request.method in ('POST', 'PUT'):
+            return request.user.is_authenticated()
         return request.method in permissions.SAFE_METHODS
 
+    def has_object_permission(self, request, view, obj):
+        userprofile = request.user.userprofile
+        return (
```

Comment:
I guesss this is to restrictive, it doesn't allow the regular user to view the offer, getOffer() returns 403.

Reply:
I believe that's a reason, that we have a lot tests failing here - I will fix that, but have stuck for few evenings trying to mock this image downlaoder.

I eventually done that but since yestarday have other idea, how it shuld work - I will today evening create pull request to separately process a change there and then will fix tests/logic here (also migrating from raw item craetion to factories, whenever I can).